src/backend/access/transam/twophase.c

```diff
@@ -197,6 +197,12 @@ static GlobalTransaction MyLockedGxact = NULL;
 
 static bool twophaseExitRegistered = false;
 
+/*
+ * xactCommitStartTime is the time at which we started to commit
+ * the current transaction.
+ */
+static instr_time xactCommitStartTime;
```

Comment:
Maybe a better place for this would be in pgstat_xact. You keep a `pgStatPreCommitTime` as a global variable and expose a `AtPreCommit_PgStat` function that would set this variable to the current time.
This way, you will just need to call `AtPreCommit_PgStat` in CommitTransaction (and the equivalent for the 2PC) and `AtEOXact_PgStat` can directly use `pgStatPreCommitTime`